letsencrypt/client.py

```diff
@@ -395,7 +396,43 @@ def redirect_to_ssl(self, domains):
                     raise
 
             self.installer.save("Add Redirects")
+
+        restart_error_handler = error_handler.ErrorHandler(
+            functools.partial(self._rollback_and_restart,
+                              self._SUCCESSFUL_REDIRECT_RECOVERY_MSG))
+        with restart_error_handler:
+            self.installer.restart()
+
+    def _recovery_routine_with_msg(self, success_msg):
+        """Calls the installer's recovery routine and prints success_msg
+
+        :param str success_msg: message to show on successful recovery
+
+        """
+        self.installer.recovery_routine()
+        reporter = zope.component.getUtility(interfaces.IReporter)
+        reporter.add_message(success_msg, reporter.HIGH_PRIORITY)
+
+    def _rollback_and_restart(self, success_msg):
+        """Rollback the most recent checkpoint and restart the webserver
+
+        :param str success_msg: message to show on successful rollback
+
+        """
+        logger.critical("Rolling back to previous server configuration...")
+        reporter = zope.component.getUtility(interfaces.IReporter)
+        try:
+            self.installer.rollback_checkpoints()
             self.installer.restart()
+        except:
+            # TODO: suggest letshelp-letsencypt here
+            reporter.add_message(
+                "An error occured and we failed to rollback your config and "
```

Comment:
Could we make a class variable of this message as well, like _SUCCESSFUL_REDIRECT_RECOVERY_MSG. That'd be a consistent way of managing the different return messages.